docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -463,8 +463,52 @@ Typically a generic proposal will have the following structure: a developer send
 * [`RemoveGenericNervousSystemFunction`](#removegenericnervoussystemfunction).
 * [`ExecuteGenericNervousSystemFunction`](#executegenericnervoussystemfunction).
 
+### Defining a generic proposal
+
+A generic proposal is defined by two parts:
+
+1. **A target method and canister** (respectively called `target_method_name` and `target_canister_id`` in the code): This is the method that will be called if this generic proposal is adopted. A community can implement any behavior in a proposal by writing it to such a target method on a canister then registering the target method in a generic proposal.
+
+2. **A validator method and canister** (respectively called `validator_method_name` and `validator_canister_id` in the code): Since the governance canister is not aware of what a generic proposal does or in which context it will be applied, it cannot validate the proposal’s payload. Therefore, to check whether a proposal’s payload is valid at proposal submission time, the SNS community must implement this validation in a separate method (this can be on the same canister as the target method or on another one). This method is then called whenever such a generic proposal is submitted.
+
+Taking this together, this is how generic proposals work: When a generic proposal is submitted to SNS governance, SNS governance calls the validator method on the validator canister to see if the payload makes sense. If this is the case, the proposal is created and can be voted on. If the proposal is adopted, then the SNS governance canister will executed the proposal by calling the target method on the target canister.
+
+Together, this is the type of a generic proposal in the code:
+
+```candid
+  type GenericNervousSystemFunction = record {
+        validator_canister_id : opt principal;
+        target_canister_id : opt principal;
+        validator_method_name : opt text;
+        target_method_name : opt text;
+    };
+```
+
+### Security considerations when designing generic proposals
+
+There are a few important, security-critical considerations to make when adding a generic proposal. We list a few recommendations here:
+
+* **The canisters where the target and validator methods are defined should be controlled by the SNS DAO.** Otherwise such a method could change the behavior or not be available without the SNS’s control. If you need to call another method consider the next point.
```

Comment:
```suggestion
* **The canisters where the target and validator methods are defined should be controlled by the SNS DAO.** Otherwise, such a method could change the behavior or not be available without the SNS’s control. If you need to call another method, consider the next point.
```